target/linux/ramips/dts/mt7620n_netgear_jwnr2010-v5.dts

```diff
@@ -0,0 +1,137 @@
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "netgear,jwnr2010-v5", "ralink,mt7620n-soc";
+	model = "Netgear JWNR2010 v5";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "jwnr2010-v5:green:power";
+			gpios = <&gpio0 21 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "jwnr2010-v5:green:wan";
+			gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan {
+			label = "jwnr2010-v5:green:wlan";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		lan1 {
+			label = "jwnr2010-v5:green:lan1";
+			gpios = <&gpio2 3 GPIO_ACTIVE_LOW>;
+		};
+
+		lan2 {
+			label = "jwnr2010-v5:green:lan2";
+			gpios = <&gpio2 2 GPIO_ACTIVE_LOW>;
+		};
+
+		lan3 {
+			label = "jwnr2010-v5:green:lan3";
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+		};
+
+		lan4 {
+			label = "jwnr2010-v5:green:lan4";
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x20000 0x3c0000>;
+			};
+
+			partition@3e0000 {
+				label = "nvram";
+				reg = <0x3e0000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@3f0000 {
+				label = "factory";
+				reg = <0x3f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x4>;
```

Comment:
Are there addresses in 0x28, 0x2e, 0x8004, 0xe000, 0xe006?

What's the address assignment on stock firmware?
https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses

Reply:
0x4 gives the correct MAC for LAN, WAN and WLAN like the stock firmware

LAN = **1D:81:28
WAN = **1D:81:29
WLAN = same as LAN @adrianschmutzler

Reply:
if 0x4 gives the correct address for WAN, then WAN can't be different from LAN.

Typically, the same addresses are stored in 0x28 and 0x2e for use with lan/wan, whereas 0x4 is meant for the Wifi calibration data. So please, just check whether anything is in the other locations stated above.

Thanks for reporting the vendor assignment BTW.

Reply:
So, where do I test the 0x28 and 0x2e, both dts and 02_network ? should I create separate wan/lan mac in 02_network ?

lan_mac=$(mtd_get_mac_binary factory 0x28)
wan_mac=$(mtd_get_mac_binary factory 0x2e) @adrianschmutzler

Reply:
https://openwrt.org/docs/guide-developer/mac.address#find_out_about_flash_locations

Reply:
Nope, no MAC at 0x28 or 0x2e. Correct MAC found in 0x0 offset (after 4 byte) and 0x80B0 (in first byte) @adrianschmutzler

Reply:
Okay, then we have to keep it as-is.

Add a comment to the commit message about the MAC addresses from stock firmware (last byte is enough) and that 0x28 and 0x2e are empty.